packages/react-examples/src/react-components/FluentUIComponents/StylingComponents.stories.mdx

```diff
@@ -0,0 +1,120 @@
+
+<!-- TODO: format as <Code /> -->
+
+```jsx
+import { makeStyles } from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: { color: 'red' },
+});
+
+function Component() {
+  const classes = useStyles();
+
+  return <div className={classes.root} />;
+}
+```
+
+### Merging component styles
+
+There are cases where you need to merge classes from multiple `useStyles` calls. **It is not possible to just concatenate the classes!**
+
+To properly merge the classes, you need to use `ax()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.
+
+When `ax()` is called, it merges all classes from top to bottom - the latter argument overwrites the previous ones (similar to `Object.assign()`).
+
+```jsx
+const useStyles = makeStyles({
+  blueBold: {
+    color: 'blue',
+    fontWeight: 'bold',
+  },
+  red: {
+    color: 'red',
+  },
+});
+
+function Component() {
+  const classes = useStyles();
+
+  const first = ax(classes.blueBold, classes.red); // { color: 'red', fontWeight: 'bold' }
+  const second = ax(classes.red, classes.blueBold); // { color: 'blue', fontWeight: 'bold' }
+}
+```
+
+### Mapping props and state to styles
+
+Both `makeStyles` and `ax` are props and state agnostic. It is developers responsibility to, based on props and state, merge and apply appropriate parts of the `useStyles` call:
+
+```jsx
+const useStyles = makeStyles({
+  root: { color: 'red' },
+  rootPrimary: { color: 'blue' },
+});
+
+function Component(props) {
+  const classes = useStyles();
+
+  return <div className={ax('ui-component', classes.root, props.primary && classes.rootPrimary)} />;
+}
+```
+
+### Applying classes passed from parent
+
+The same approach is used to apply classes passed from parent component. Again, you need to use `ax()` to properly merge and deduplicate the classes.
+
+```jsx
+function Component(props) {
+  const classes = useStyles();
+
+  return <div className={ax(classes.root, props.className /* these definitions have higher precedence */)} />;
+}
+```
+
+### Applying theme to styles
+
+No matter what theme is used, the component styles are always the same. The only way how to change the component appearance is through tokens which can be used in style values.
+
+```js
+const useStyles = makeStyles({
+  root: { display: 'flex' },
+  rootPrimary: (theme) => ({ color: theme.alias.color.neutral.neutralForeground3 }),
+});
+```
+
+Those tokens are resolved to CSS variable usages. `ThemeProvider` component is responsible for setting the CSS variables in DOM and changing them when theme changes. When theme is switched, only the variables are changed, all styles remain the same.
+
+## Overriding FUI component styles
+
+To override an appearance of a FUI component, you use the exactly same approach - You call makeStyles/useStyles in your code and pass the resulting classes through `props`.
```

Comment:
should we add an example for shorthand slots too ? `icon` slot on `Button` should serve this purpose nicely

Reply:
Icon overrides added